src/runtime/virtcontainers/clh.go

```diff
@@ -255,12 +261,16 @@ func (clh *cloudHypervisor) CreateVM(ctx context.Context, id string, network Net
 
 	if clh.state.PID > 0 {
 		clh.Logger().WithField("function", "CreateVM").Info("Sandbox already exist, loading from state")
-		clh.virtiofsDaemon = &virtiofsd{
-			PID:        clh.state.VirtiofsDaemonPid,
-			sourcePath: hypervisorConfig.SharedPath,
-			debug:      clh.config.Debug,
-			socketPath: virtiofsdSocketPath,
+
+		if clh.supportsSharedFS() {
+			clh.virtiofsDaemon = &virtiofsd{
+				PID:        clh.state.VirtiofsDaemonPid,
+				sourcePath: hypervisorConfig.SharedPath,
+				debug:      clh.config.Debug,
+				socketPath: virtiofsdSocketPath,
+			}
```

Comment:
I feel this adds a bit of complexity to the code and makes it slightly less easy to read/review. It could be worth defining a set of `init/start/stop` routines for managing the virtiofsd lifecycle and include the `supportsSharedFS()` call in there. This would make for a more readable code imho.
But that's not a show stopper for that PR.

Reply:
Addressed!